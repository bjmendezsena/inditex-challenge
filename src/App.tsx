import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { persistQueryClient } from "@tanstack/react-query-persist-client";
import { createSyncStoragePersister } from "@tanstack/query-sync-storage-persister";
import { theme } from "./theme";
import Router from "./router";
import { CACHE_TIME } from "./constants";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: CACHE_TIME,
      refetchOnWindowFocus: false,
    },
  },
});

const localStoragePersister = createSyncStoragePersister({
  storage: localStorage,
});

persistQueryClient({
  queryClient,
  persister: localStoragePersister,
});

const App = () => {
  return (
    <ChakraProvider theme={theme}>
      <QueryClientProvider client={queryClient}>
        <Router />
      </QueryClientProvider>
    </ChakraProvider>
  );
};

export default App;
