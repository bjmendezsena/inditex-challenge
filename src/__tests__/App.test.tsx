import { render } from "@testing-library/react";
import App from "../App";

describe(`<${App.name}/>`, () => {
  it("should render without crash", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
