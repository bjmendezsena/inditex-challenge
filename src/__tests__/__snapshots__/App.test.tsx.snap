// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render without crash 1`] = `
<DocumentFragment>
  <h2>
    Unexpected Application Error!
  </h2>
  <h3
    style="font-style: italic;"
  >
    Cannot read properties of undefined (reading 'im:image')
  </h3>
  <pre
    style="padding: 0.5rem; background-color: rgba(200, 200, 200, 0.5);"
  >
    TypeError: Cannot read properties of undefined (reading 'im:image')
    at map (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/src/components/Organisms/PodcastList/PodcastList.tsx:38:24)
    at Array.map (&lt;anonymous&gt;)
    at PodcastList (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/src/components/Organisms/PodcastList/PodcastList.tsx:28:14)
    at renderWithHooks (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:16305:18)
    at mountIndeterminateComponent (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:20074:13)
    at beginWork (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:21587:16)
    at beginWork$1 (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:27426:14)
    at performUnitOfWork (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:26560:12)
    at workLoopSync (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:26466:5)
    at renderRootSync (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:26434:7)
    at recoverFromConcurrentError (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:25850:20)
    at performConcurrentWorkOnRoot (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react-dom/cjs/react-dom.development.js:25750:22)
    at flushActQueue (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react/cjs/react.development.js:2667:24)
    at act (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/react/cjs/react.development.js:2582:11)
    at /Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/@testing-library/react/dist/act-compat.js:63:25
    at renderRoot (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/@testing-library/react/dist/pure.js:159:26)
    at render (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/@testing-library/react/dist/pure.js:246:10)
    at Object.&lt;anonymous&gt; (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/src/__tests__/App.test.tsx:6:38)
    at Promise.then.completed (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/utils.js:391:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/run.js:218:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at _runTest (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/run.js:60:9)
    at run (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
    at jestAdapter (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
    at runTestInternal (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-runner/build/runTest.js:389:16)
    at runTest (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-runner/build/runTest.js:475:34)
    at Object.worker (/Users/briaulinmendez/Desktop/pruebas_tecnicas/inditex-challenge/node_modules/jest-runner/build/testWorker.js:133:12)
  </pre>
  <p>
    ðŸ’¿ Hey developer ðŸ‘‹
  </p>
  <p>
    You can provide a way better UX than this when your app throws errors by providing your own 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      ErrorBoundary
    </code>
     or 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      errorElement
    </code>
     prop on your route.
  </p>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
